pkg/controller/clusterinstallation/controller.go

```diff
@@ -118,27 +109,23 @@ func (r *ReconcileClusterInstallation) Reconcile(request reconcile.Request) (rec
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 	reqLogger.Info("Reconciling ClusterInstallation")
 
-	// Fetch the ClusterInstallation instance
+	// Fetch the ClusterInstallation.
 	mattermost := &mattermostv1alpha1.ClusterInstallation{}
 	err := r.client.Get(context.TODO(), request.NamespacedName, mattermost)
 	if err != nil && k8sErrors.IsNotFound(err) {
-		// Request object not found, could have been deleted after reconcile request.
-		// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-		// Return and don't requeue
-		r.setReconciling()
+		// Request object not found, could have been deleted after reconcile
+		// request. Owned objects are automatically garbage collected. For
+		// additional cleanup logic use finalizers. Return and don't requeue.
 		return reconcile.Result{}, nil
 	} else if err != nil {
 		// Error reading the object - requeue the request.
-		r.setReconciling()
 		return reconcile.Result{}, err
 	}
 
-	if mattermost.Status.State != r.state {
-		status := mattermost.Status
-		status.State = r.state
-		err = r.updateStatus(mattermost, status, reqLogger)
+	// Set a new ClusterInstallation's state to reconciling.
+	if len(mattermost.Status.State) == 0 {
+		err = r.setStateReconciling(mattermost, reqLogger)
 		if err != nil {
-			r.setReconciling()
 			return reconcile.Result{}, err
 		}
```

Comment:
Just one question: Is there a reason not to log the error here?
I think it was previously logged inside `updateStatus` but as it is no longer the case, you might have missed it.

Reply:
Good question! The returned errors from the reconcile function are always logged so we just let that handle it. In this case the error is that the status could not be set, but in other places we need to log that error separately.